nifi-bootstrap/src/main/java/org/apache/nifi/bootstrap/process/AvailablePorts.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.bootstrap.process;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class AvailablePorts implements RuntimeValidator {
+    private static final String DIRECTORY = "/proc/sys/net/ipv4/ip_local_port_range";
+    private static final String DIGITS_REGEX = "(\\d+)\\s+(\\d+)";
+    private static final Pattern PATTERN = Pattern.compile(DIGITS_REGEX);
+    private static final int DESIRED_AVAILABLE_PORTS = 55000;
+
+    private final File configurationFile;
+
+    public AvailablePorts() {
+        configurationFile = new File(DIRECTORY);
+    }
+
+    AvailablePorts(final File configurationFile) {
+        this.configurationFile = configurationFile;
+    }
+
+    @Override
+    public List<RuntimeValidatorResult> check() {
+        final List<RuntimeValidatorResult> results = new ArrayList<>();
+        if (configurationFile == null) {
+            final RuntimeValidatorResult result = new RuntimeValidatorResult.Builder()
+                    .subject(this.getClass().getName())
+                    .outcome(RuntimeValidatorResult.Outcome.SKIPPED)
+                    .explanation("Configuration file is null")
+                    .build();
+            results.add(result);
+            return results;
+        }
+        if (!configurationFile.canRead()) {
+            final RuntimeValidatorResult result = new RuntimeValidatorResult.Builder()
+                    .subject(this.getClass().getName())
+                    .outcome(RuntimeValidatorResult.Outcome.SKIPPED)
+                    .explanation(String.format("Configuration file [%s] cannot be read", configurationFile.getAbsolutePath()))
+                    .build();
+            results.add(result);
+            return results;
+        }
```

Comment:
This file existence checking looks similar across implementations, so it would be helpful to pull it out to a shared abstract class.